### ISSUE TITLE
Spec failing to lint when pulling in static framework as dependency

### ISSUE THREAD

**Issue by 0xTim** [CocoaPods/CocoaPods#8781]
TL;DR Our project used to work on 1.5.3, when we updated to 1.6.1 the Podspec failed to lint with no code changes (we build daily on CI to stop build rot).

# Report

## What did you do?

Ran `pod lib lint`

## What did you expect to happen?

Podspec to lint correctly

## What happened instead?

Target failed to link with undefined symbols. Full build output:

```
Undefined symbols for architecture x86_64:
      "_OBJC_CLASS_$_GCKUICastButton", referenced from:
          objc-class-ref in GCKUICastButton+BBCCTDefaultCastButton.o
          l_OBJC_$_CATEGORY_GCKUICastButton_$_BBCCTDefaultCastButton in GCKUICastButton+BBCCTDefaultCastButton.o
      "_OBJC_CLASS_$_GCKRemoteMediaClient", referenced from:
          l_OBJC_$_CATEGORY_GCKRemoteMediaClient_$_BBCCTCastMediaDetails in GCKRemoteMediaClient+BBCCTCastMediaDetails.o
      "_OBJC_CLASS_$_GCKMediaLoadOptions", referenced from:
          objc-class-ref in GCKRemoteMediaClient+BBCCTCastMediaDetails.o
      "_OBJC_CLASS_$_GCKMediaStatus", referenced from:
          l_OBJC_$_CATEGORY_GCKMediaStatus_$_Subtitles in GCKMediaStatus+Subtitles.o
      "_OBJC_CLASS_$_GCKCastContext", referenced from:
          objc-class-ref in GCKCastContext+BBCCTDefaultCastContext.o
          l_OBJC_$_CATEGORY_GCKCastContext_$_BBCCTDefaultCastContext in GCKCastContext+BBCCTDefaultCastContext.o
      "_OBJC_CLASS_$_GCKDiscoveryCriteria", referenced from:
          objc-class-ref in GCKCastContext+BBCCTDefaultCastContext.o
      "_kGCKMetadataKeySubtitle", referenced from:
          -[BBCCTCastMediaDetailsMapper map:] in BBCCTCastMediaDetailsMapper.o
          -[BBCCTCastMediaDetailsMapper mapMediaInformation:] in BBCCTCastMediaDetailsMapper.o
      "_OBJC_CLASS_$_GCKUIExpandedMediaControlsViewController", referenced from:
          objc-class-ref in BBCCTUIStyler.o
      "_OBJC_CLASS_$_GCKCastChannel", referenced from:
          objc-class-ref in BBCCTSubtitlesManager.o
      "_OBJC_CLASS_$_GCKMediaInformation", referenced from:
          objc-class-ref in BBCCTCastMediaDetailsMapper.o
          l_OBJC_$_CATEGORY_GCKMediaInformation_$_Subtitles in GCKMediaInformation+Subtitles.o
      "_OBJC_CLASS_$_GCKUIStyle", referenced from:
          objc-class-ref in BBCCTCast.o
      "_OBJC_CLASS_$_GCKMediaMetadata", referenced from:
          objc-class-ref in BBCCTCastMediaDetailsMapper.o
      "_OBJC_CLASS_$_GCKImage", referenced from:
          objc-class-ref in BBCCTCastMediaDetailsMapper.o
      "_OBJC_CLASS_$_GCKCastOptions", referenced from:
          objc-class-ref in GCKCastContext+BBCCTDefaultCastContext.o
      "_kGCKMetadataKeyTitle", referenced from:
          -[BBCCTCastMediaDetailsMapper map:] in BBCCTCastMediaDetailsMapper.o
          -[BBCCTCastMediaDetailsMapper mapMediaInformation:] in BBCCTCastMediaDetailsMapper.o
    ld: symbol(s) not found for architecture x86_64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    
    ** BUILD FAILED **
```

### Stack

```
   CocoaPods : 1.6.1
        Ruby : ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-darwin18]
    RubyGems : 2.7.6
        Host : Mac OS X 10.14.4 (18E226)
       Xcode : 10.2.1 (10E1001)
         Git : git version 2.21.0
Ruby lib dir : /Users/timc/.rbenv/versions/2.5.3/lib
Repositories : bbc - git@github.com:bbc/map-ios-podspecs.git @ 0fc8622d198e8c9c3206b515a4db7925c51fff28
               master - https://github.com/CocoaPods/Specs.git @ 70d7091d18b5a15ee8e95faec006647dfec2f23b
```

### Installation Source

```
Executable Path: /Users/timc/.rbenv/versions/2.5.3/bin/pod
```

### Plugins

```
cocoapods-acknowledgements : 1.1.3
cocoapods-deintegrate      : 1.0.4
cocoapods-plugins          : 1.0.0
cocoapods-search           : 1.0.0
cocoapods-stats            : 1.1.0
cocoapods-trunk            : 1.3.1
cocoapods-try              : 1.1.0
```

### Podfile

```ruby
platform :ios, '9.0'
use_frameworks!
inhibit_all_warnings!

source 'https://github.com/CocoaPods/Specs.git'

target 'CastToolkit' do
	podspec

    target 'CastToolkitTests' do
        pod 'OCMock', '3.3.1'
    end
end
```

## Project that demonstrates the issue

I've recreated a sample project to demonstrate the problem - https://github.com/0xTim/cast-toolkit-pod-lint-failure

Running `pod lib lint --allow-warnings --no-clean` produces a failure with a workspace that has similar errors to those posted above.
**Note:** we have managed to workaround the issue using `s.static_framework = true`. However since this was introduced in 1.4.0 and only broke in 1.6.1 I'm not sure if this is an intended change of behaviour (to force Podspecs to include this) or a regression between 1.5.3 and 1.6.1. Not sure about the impact of setting this flag yet and still waiting for testing to complete in downstream projects that depend upon the library.

**Comment by joyl1216** [comment]
Got the same error after moving from 1.5.2 to 1.6.1, and even 1.6.2.

**Comment by paulb777** [comment]
I can't comment on whether this change was intended.

However, it is definitely strongly recommended to use `s.static_framework = true` for pods that depend on static frameworks, since otherwise apps will get undefined behavior if more than one of their dependencies depends upon a static framework.

**Comment by joyl1216** [comment]
@paulb777 You are right. `use_frameworks!` is specified in my Podfile. Uncomment it will fix the issue in `1.7.0.rc.2`. But in `1.6.2` I got another error `:-1: Unable to load contents of file list: 'xxxxx/Pods/Target Support Files/Pods-xxxx/Pods-xxxxx-frameworks-Debug-input-files.xcfilelist' (in target 'xxxxx')`.
